### PR TITLE
maintainer scripts: init check-hydra-by-maintainer.nix

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change

There was no easy way to do this.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Review by siraben — APPROVED** [review]
LGTM. Tried it on myself, it was a bit slow probably due to hydra rate limiting requests. Looks like the `packagesWith` function is copied from `maintainers/scripts/build.nix`, perhaps eventually we could have some standard library for querying package metadata.

<details>

```
$ nix-shell maintainers/scripts/check-hydra-by-maintainer.nix --argstr maintainer siraben         
nix-shell -p hydra-check --run hydra-check cyclone-scheme knightos-kcc knightos-kimg knightos-kpack knightos-mkrom knightos-mktiupgrade knightos-patchrom knightos-scas knightos-z80e miranda mozwire python37Packages.swspotify python38Packages.swspotify swaglyrics the-powder-toy tilem
Build Status for nixpkgs.cyclone-scheme.x86_64-linux on unstable
✔ cyclone-0.21 from 2020-10-17 - https://hydra.nixos.org/build/128626648
Build Status for nixpkgs.knightos-kcc.x86_64-linux on unstable
✔ kcc-4.0.0 from 2020-10-17 - https://hydra.nixos.org/build/128637613
Build Status for nixpkgs.knightos-kimg.x86_64-linux on unstable
✔ kimg-0.3.0 from 2020-10-25 - https://hydra.nixos.org/build/129096700
Build Status for nixpkgs.knightos-kpack.x86_64-linux on unstable
✔ kpack-1.1.0 from 2020-10-17 - https://hydra.nixos.org/build/128629769
Build Status for nixpkgs.knightos-mkrom.x86_64-linux on unstable
✔ mkrom-1.0.2 from 2020-10-25 - https://hydra.nixos.org/build/129096239
Build Status for nixpkgs.knightos-mktiupgrade.x86_64-linux on unstable
✔ mktiupgrade-1.1.6 from 2020-10-25 - https://hydra.nixos.org/build/129096997
Build Status for nixpkgs.knightos-patchrom.x86_64-linux on unstable
✔ patchrom-1.1.3 from 2020-10-18 - https://hydra.nixos.org/build/128639452
Build Status for nixpkgs.knightos-scas.x86_64-linux on unstable
✔ scas-0.4.6 from 2020-10-17 - https://hydra.nixos.org/build/128617802
Build Status for nixpkgs.knightos-z80e.x86_64-linux on unstable
✔ z80e-0.5.0 from 2020-10-18 - https://hydra.nixos.org/build/128679490
Build Status for nixpkgs.miranda.x86_64-linux on unstable
✔ miranda-2.066 from 2020-10-18 - https://hydra.nixos.org/build/128625636
Build Status for nixpkgs.mozwire.x86_64-linux on unstable
✔ MozWire-0.5.2 from 2020-10-17 - https://hydra.nixos.org/build/128618575
Build Status for nixpkgs.python37Packages.swspotify.x86_64-linux on unstable
✔ python3.7-SwSpotify-1.2.1 from 2020-10-18 - https://hydra.nixos.org/build/128632961
Build Status for nixpkgs.python38Packages.swspotify.x86_64-linux on unstable
✔ python3.8-SwSpotify-1.2.1 from 2020-10-18 - https://hydra.nixos.org/build/128615712
Build Status for nixpkgs.swaglyrics.x86_64-linux on unstable
✔ swaglyrics-1.2.2 from 2020-10-18 - https://hydra.nixos.org/build/128622615
Build Status for nixpkgs.the-powder-toy.x86_64-linux on unstable
✔ the-powder-toy-95.0 from 2020-10-17 - https://hydra.nixos.org/build/128624368
Build Status for nixpkgs.tilem.x86_64-linux on unstable
✔ tilem-2.0 from 2020-10-22 - https://hydra.nixos.org/build/128854301
...
```
</details>

**Comment by SuperSandro2000** [comment]
> Looks like the packagesWith function is copied from maintainers/scripts/build.nix, perhaps eventually we could have some standard library for querying package metadata.

Thats true. My Nix is not that good yet.
I am open to improvements.

**Comment by Ma27** [comment]
(sidenote: as soon as https://github.com/NixOS/hydra/pull/743 is merged, we could simplify this. Unfortunately I need to find some time to implement a proper DB migration here first)

**Comment by Mic92** [comment]
> LGTM. Tried it on myself, it was a bit slow probably due to hydra rate limiting requests. Looks like the `packagesWith` function is copied from `maintainers/scripts/build.nix`, perhaps eventually we could have some standard library for querying package metadata.

I think hydra got a bit faster lately.

**Comment by Ma27** [comment]
So I'm sorry if I wasn't sufficiently clear last time, my PR in Hydra *shouldn't* block this PR. IIRC I briefly tested it last time and it appeared fine to me. So if someone else approves it, it should be good to go.

**Review by 0x4A6F — APPROVED** [review]
Thanks for your work.

Tested:
- [x] nix-shell maintainers/scripts/check-hydra-by-maintainer.nix --argstr maintainer _0x4A6F

**Pull request merged**
